libasherah.go

```diff
@@ -321,24 +249,24 @@ func Decrypt(partitionIdPtr unsafe.Pointer, encryptedDataPtr unsafe.Pointer, enc
 	globalDebugOutput("Decrypt()")
 
 	partitionId, result := cobhan.BufferToString(partitionIdPtr)
-	if result != 0 {
+	if result != ERR_NONE {
 		return result
 	}
 
 	globalDebugOutput("Decrypting with partition: " + partitionId)
 
 	encryptedData, result := cobhan.BufferToBytes(encryptedDataPtr)
-	if result != 0 {
+	if result != ERR_NONE {
 		return result
 	}
 
 	encryptedKey, result := cobhan.BufferToBytes(encryptedKeyPtr)
-	if result != 0 {
+	if result != ERR_NONE {
 		return result
 	}
 
 	parentKeyId, result := cobhan.BufferToString(parentKeyIdPtr)
-	if result != 0 {
+	if result != ERR_NONE {
```

Comment:
Should we generalize all these cobhan error codes under a single `ERR_CONFIGUGARION_FAILED` to prevent propagating cobhan error codes to asherah consumers or have them defined in asherah as well so they get more visibility? Similar for Encrypt/Decrypt functions below.